client/src/components/ui/TermsAndConditionsModal.tsx

```diff
@@ -0,0 +1,100 @@
+import { useLocalize } from '~/hooks';
+import { Dialog } from '~/components/ui';
+import DialogTemplate from '~/components/ui/DialogTemplate';
+import { useAuthContext } from '~/hooks';
+import Markdown from '~/components/Chat/Messages/Content/Markdown';
+
+const TermsAndConditionsModal = ({
+  open,
+  onOpenChange,
+  onAccept,
+  onDecline,
+  title,
+  modalContent,
+}: {
+  open: boolean;
+  onOpenChange: (isOpen: boolean) => void;
+  onAccept: () => void;
+  onDecline: () => void;
+  title?: string;
+  contentUrl?: string;
+  modalContent?: string;
+}) => {
+  const localize = useLocalize();
+  const { token } = useAuthContext();
+
+  const handleAccept = async () => {
+    try {
+      const response = await fetch('/api/user/terms/accept', {
```

Comment:
don't make raw requests, use react query (in `data-provider`). This apply for both the post and get request